ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
+        }
+
         $button = array(
           array(
             '#theme' => 'link',
             '#text' => t('Reserve'),
-            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . implode('/', $path),
+            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . base64_encode(implode('|', $item_id)),
```

Comment:
The reason for this approach is that having "/" in the item id is confusing drupal routing, therefore considering the resulting path to be treated as whole.

There were attempts using urlencode/rawurlencode yet both of them resulted in fail to reserve a certain periodical. Since periodicals that we encountered had not only slashes in it;s id, but also ampersands and dots.
base64_encode was the only viable way to encode everything into alpha-numeric symbols. So no special characters are passed to the hook_menu argument.